core/src/types.rs

```diff
@@ -243,6 +246,7 @@ impl FromStr for ModelSource {
         match s.to_lowercase().as_str() {
             "openai" => Ok(ModelSource::OpenAI),
             "sentence-transformers" => Ok(ModelSource::SentenceTransformers),
+            "llama2" => Ok(ModelSource::Ollama),
```

Comment:
Would this be "ollama", then a use can request to use llama2 by providing `chat_model => 'ollama/lamma2'`?